plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/exporter/QueryInsightsLocalIndexExporter.java

```diff
@@ -0,0 +1,216 @@
+                        }
+                    } else {
+                        log.error(
+                            String.format(
+                                Locale.ROOT,
+                                "request to created local index %s for query insight not acknowledged.",
+                                getIdentifier()
+                            )
+                        );
+                    }
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    log.error(String.format(Locale.ROOT, "error creating local index for query insight: %s", e));
+                }
+            });
+        }
+    }
+
+    /**
+     * Util function to check if a local OpenSearch Index exists
+     *
+     * @return boolean
+     */
+    private boolean checkIfIndexExists() {
+        ClusterState clusterState = clusterService.state();
+        return clusterState.getRoutingTable().hasIndex(this.getIdentifier());
+    }
+
+    /**
+     * Initialize the local OpenSearch Index for the exporter
+     *
+     * @param listener the listener to be notified upon completion
+     * @throws IOException if an error occurs
+     */
+    private synchronized void initLocalIndex(ActionListener<CreateIndexResponse> listener) throws IOException {
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest(this.getIdentifier()).mapping(getIndexMappings())
+            .settings(Settings.builder().put("index.hidden", false).build());
+        client.admin().indices().create(createIndexRequest, listener);
+    }
+
+    /**
+     * Drop the local OpenSearch Index created by the exporter
+     *
+     * @param listener the listener to be notified upon completion
+     * @throws IOException if an error occurs
+     */
+    private synchronized void dropLocalIndex(ActionListener<AcknowledgedResponse> listener) throws IOException {
+        DeleteIndexRequest deleteIndexRequest = new DeleteIndexRequest(this.getIdentifier());
+        client.admin().indices().delete(deleteIndexRequest, listener);
+    }
+
+    /**
+     * Get the index mapping of the local index
+     *
+     * @return String to represent the index mapping
+     * @throws IOException if an error occurs
+     */
+    private String getIndexMappings() throws IOException {
+        return new String(Objects.requireNonNull(this.localIndexMapping).readAllBytes(), Charset.defaultCharset());
+    }
+
+    /**
+     * Bulk ingest the data into to the predefined local OpenSearch Index
+     *
+     * @param records the data to export
+     * @throws IOException if an error occurs
+     */
+    private synchronized void bulkRecord(List<T> records) throws IOException {
+        BulkRequest bulkRequest = new BulkRequest().setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+            .timeout(TimeValue.timeValueSeconds(INDEX_TIMEOUT));
```

Comment:
Does this method really need to be synchronized?

Reply:
Good point, removing this